pkg/ipamclaimswebhook/podmutator.go

```diff
@@ -233,54 +276,94 @@ func ensureIPAMClaimRefAtNetworkSelectionElements(ctx context.Context,
+func findPrimaryUDNLogicalNetworkAttachment(ctx context.Context,
+	cli client.Client, vmi *virtv1.VirtualMachineInstance) (*virtv1.Interface, error) {
 	log := logf.FromContext(ctx)
-	if pod.Annotations[config.OVNPrimaryNetworkIPAMClaimAnnotation] != "" {
-		return "", nil
-	}
-	primaryNetworkNAD, err := udn.FindPrimaryNetwork(ctx, cli, pod.Namespace)
+	primaryNetworkNAD, err := udn.FindPrimaryNetwork(ctx, cli, vmi.Namespace)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	if primaryNetworkNAD == nil {
-		return "", nil
+		return nil, nil
 	}
 	pluginConfig, err := config.NewConfig(primaryNetworkNAD.Spec.Config)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
 	if !pluginConfig.AllowPersistentIPs {
-		return "", nil
+		return nil, nil
 	}
 
 	log.Info(
 		"will request primary network persistent IPs",
 		"NAD", client.ObjectKeyFromObject(primaryNetworkNAD),
 		"network", pluginConfig.Name,
 	)
+
+	podNetwork := vmiPodNetwork(vmi)
+	if podNetwork == nil {
+		log.Info(
+			"vmi has no pod network primary UDN ipam claim will not be created",
+			"vmi",
+			client.ObjectKeyFromObject(vmi),
+		)
+		return nil, nil
+	}
+
+	return vmiPodPrimaryNetworkInterface(vmi, *podNetwork), nil
+}
+
+func runningVMForPod(pod *corev1.Pod, vmName string, cli client.Client) (*virtv1.VirtualMachineInstance, error) {
 	vmKey := types.NamespacedName{Namespace: pod.Namespace, Name: vmName}
 	vmi := &virtv1.VirtualMachineInstance{}
 	if err := cli.Get(context.Background(), vmKey, vmi); err != nil {
-		return "", err
+		return nil, err
 	}
-
-	networkName := vmiPodNetworkName(vmi)
-	if networkName == "" {
-		log.Info("vmi has no pod network primary UDN ipam claim will not be created", "vmi", vmKey.String())
-		return "", nil
-	}
-
-	return claims.ComposeKey(vmi.Name, networkName), nil
+	return vmi, nil
 }
 
-// returns the name of the kubevirt VM pod network
-func vmiPodNetworkName(vmi *virtv1.VirtualMachineInstance) string {
+// returns the KubeVirt VM pod network
+func vmiPodNetwork(vmi *virtv1.VirtualMachineInstance) *virtv1.Network {
 	for _, network := range vmi.Spec.Networks {
 		if network.Pod != nil {
-			return network.Name
+			return &network
 		}
 	}
-	return ""
+	return nil
+}
+
+func vmiPodPrimaryNetworkInterface(vmi *virtv1.VirtualMachineInstance, network virtv1.Network) *virtv1.Interface {
+	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
+		if iface.Name == network.Name {
+			return &iface
+		}
+	}
+	return nil
+}
+
+func multusDefaultNetworkAnnotation(
+	namespace string,
+	mac string,
+	ipamClaimName string,
+	ips ...string,
+) *v1.NetworkSelectionElement {
+	const defaultNetworkName = "default"
+	return &v1.NetworkSelectionElement{
+		Name:               defaultNetworkName,
+		Namespace:          namespace,
+		IPRequest:          ips,
+		MacRequest:         mac,
+		IPAMClaimReference: ipamClaimName,
+	}
+}
+
+func primaryNetworkIPRequests(vmi *virtv1.VirtualMachineInstance, podNetworkName string) []string {
```

Comment:
We should name things for what it does, the "primary" part is what the caller is using it for.
```suggestion
func vmiInterfaceIPRequests(vmi *virtv1.VirtualMachineInstance, vmiInterfaceName string) []string {
```